### DISCUSSION TITLE
Dispatch failure

### DISCUSSION THREAD

**Discussion by DoIevenLift** [cargo-lambda/cargo-lambda#701] [Q&A]
Hi, 

I am wondering if anyone has encountered this issue when doing cargo lambda deploy <fn_name> --profile <user>:

![image](https://github.com/user-attachments/assets/5a2be91c-79c1-4a23-ae83-c6627c018fa4)

If anyone has any pointers on how to resolve that'd be great!

Thanks!

**Comment by DoIevenLift** [comment]
Just adding some more context here:

using -v I get this:

![image](https://github.com/user-attachments/assets/1420452d-022e-41b2-9769-dce67f2b3345)

Which I assume is the cli searching for the lambda function? The function definitely exists and was deployed last month using cargo lambda deploy.

I've also tried to reset my credentials via aws config and doing this manually as well which is returning the same error. 

Also tried this on another machine with the same aws credentials and got the same error as well. 

Thanks

**Reply by calavera** [reply]
`error trying to connect: invalid peer certificate: UnknownIssuer` comes from the AWS Lambda control plane. The SDK is trying to use an invalid TLS certificate to establish the connection between your computer and AWS. I don't know how that's possible, but it's probably not related to Cargo Lambda.

**Reply by DoIevenLift** [reply]
Hi @calavera , 

Thanks for the reply. I have no idea either! Odd that its happening across two machines - will keep searching!

**Reply by DoIevenLift** [reply]
Hi @calavera, 

I did a complete reinstall of windows for a clean PC the only things installed have been rust, cargo lambda, zig, visual studio stuff and mingw. Continue to get the error. 

Do you / can you think of any other place to point me? Happy to continue to explore but this is well above my skill set / knowledge. 

Thanks

**Reply by DoIevenLift** [reply]
sorry for a bit of spam lately - just been focusing on trying to resolve this. 

I've tested with the aws cli directly and can interact with aws lambda and do actions like list, create, delete etc. According to this - [https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/index.html#description](url) - there are CA certs required. I've confirmed on both machines that these exist and even checked a 3rd entirely separate one and these have the same certs. 

If this was a certificate issue the aws cli commands should also return the same error right? So far I am only experiencing this with cargo lambda commands. If there is anything else I can try let me know. 

Thanks again and apologies (again) about the spam.

**Reply by calavera** [reply]
That's very strange indeed. How are you installing Cargo Lambda? What does your environment look like? The AWS SDK should take those certificates in the same way the aws cli does.

**Reply by DoIevenLift** [reply]
Installing cargo lambda via scoop commands and aws cli via the installer. For the env this is a windows machine with nothing special.

**Reply by calavera** [reply]
I found this old reference to a similar issue, but I don't know why your system can be affected by it: https://github.com/awslabs/aws-sdk-rust/issues/468#issuecomment-1067017008

**Reply by DoIevenLift** [reply]
I'll make an issue / discussion on aws-sdk and see. Very **very** odd that this is happening across multiple machines.

**Reply by DoIevenLift** [reply]
I have found the solution to this.

For whatever reason in this case, if the AWS CLI command is used the certificate will validate. When using the Cargo Lambda commands, the certificate used would not validate and return the error. 

Downloading any of the Amazon CA certs from - [https://www.amazontrust.com/repository/](https://www.amazontrust.com/repository/) and importing this via the Windows wizard into the Trusted Root CA then generates the CA twice - its like it unhides / marks as trusted the original CA that existed. This solves the issue and allows the Lambda console to approve the request.

While this solves the problem it doesn't 'explain' how / why the Cargo Lambda commands are not validating the existing certificate like the AWS CLI commands are.

This may also be unrelated to Cargo Lambda and could be some magic that Windows did with an update which either untrusted an Amazon CA or something but there 'aren't' any other issues / discussions seen in these channels so far (which makes this potentially unlikely) and debugging further here is way above my skill set.

Really appreciate the help and your time!

Thanks

**Reply by calavera** [reply]
I'm glad you were able to figure it out!
